### PR TITLE
qt: Remove obsolete QModelIndex::child()

### PR DESCRIPTION
**Pull request by hebasto**
The `QModelIndex::child()` member function is [obsolete](https://doc.qt.io/qt-5.12/qmodelindex-obsolete.html) since Qt 5.12.

This PR removes it, does not change behavior and keeps compatibility with [Qt 5.5.1](https://github.com/bitcoin/bitcoin/pull/15393).

Here is an excerpt from the master build log:
```
qt/receivecoinsdialog.cpp: In member function ‘void ReceiveCoinsDialog::copyColumnToClipboard(int)’:
qt/receivecoinsdialog.cpp:264:111: warning: ‘QModelIndex QModelIndex::child(int, int) const’ is deprecated: Use QAbstractItemModel::index [-Wdeprecated-declarations]
     GUIUtil::setClipboard(model->getRecentRequestsTableModel()->data(firstIndex.child(firstIndex.row(), column), Qt::EditRole).toString());
                                                                                                               ^
In file included from /home/hebasto/Qt/5.13.0/gcc_64/include/QtWidgets/qabstractitemview.h:45:0,
                 from /home/hebasto/Qt/5.13.0/gcc_64/include/QtWidgets/qheaderview.h:44,
                 from /home/hebasto/Qt/5.13.0/gcc_64/include/QtWidgets/QHeaderView:1,
                 from ./qt/guiutil.h:12,
                 from ./qt/receivecoinsdialog.h:8,
                 from qt/receivecoinsdialog.cpp:7:
/home/hebasto/Qt/5.13.0/gcc_64/include/QtCore/qabstractitemmodel.h:457:20: note: declared here
 inline QModelIndex QModelIndex::child(int arow, int acolumn) const
                    ^~~~~~~~~~~
```

### CODE REVIEW

File: src/qt/recentrequeststablemodel.cpp [hunk 1]
```diff
@@ -125,7 +125,7 @@ QString RecentRequestsTableModel::getAmountTitle()
     return (this->walletModel->getOptionsModel() != nullptr) ? tr("Requested") + " ("+BitcoinUnits::shortName(this->walletModel->getOptionsModel()->getDisplayUnit()) + ")" : "";
 }
 
-QModelIndex RecentRequestsTableModel::index(int row, int column, const QModelIndex &parent) const
```
Comment (promag) [inline src/qt/recentrequeststablemodel.cpp]: nit, drop unnecessary change?

File: src/qt/receivecoinsdialog.cpp [hunk 2]
```diff
@@ -261,7 +261,9 @@ void ReceiveCoinsDialog::copyColumnToClipboard(int column)
     if (!firstIndex.isValid()) {
         return;
     }
-    GUIUtil::setClipboard(model->getRecentRequestsTableModel()->data(firstIndex.child(firstIndex.row(), column), Qt::EditRole).toString());
+
+    const RecentRequestsTableModel* const submodel = model->getRecentRequestsTableModel();
+    GUIUtil::setClipboard(submodel->data(submodel->index(firstIndex.row(), column), Qt::EditRole).toString());
```
Comment (promag) [inline src/qt/receivecoinsdialog.cpp]: I think you could do
```cpp
 GUIUtil::setClipboard(model->getRecentRequestsTableModel()->index(firstIndex.row(), column).data(Qt::EditRole).toString());
```

**Review by promag — COMMENTED** [review]
Concept ACK.

File: src/qt/recentrequeststablemodel.cpp (hunk 1 above)
Comment (hebasto) [reply src/qt/recentrequeststablemodel.cpp]: Done.

File: src/qt/receivecoinsdialog.cpp (hunk 2 above)
Comment (hebasto) [reply src/qt/receivecoinsdialog.cpp]: Done.

**Comment by hebasto** [comment]
@promag
Thank you for your review. All your comments have been addressed.

**Comment by laanwj** [comment]
Code review ACK c53667764ea65c0f8661358759b0124f808ce587

**Comment by promag** [comment]
ACK c53667764ea65c0f8661358759b0124f808ce587, just read the change.

**Review by jonasschnelli — APPROVED** [review]
utACK c53667764ea65c0f8661358759b0124f808ce587
